src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -365,8 +365,11 @@ internal override void ForceComplete(SourceLocation locationOpt, Predicate<Symbo
                         // Nested type parameter references might not be valid in error scenarios.
                         //Debug.Assert(this.ContainingSymbol.IsContainingSymbolOfAllTypeParameters(this.ConstraintTypes));
                         //Debug.Assert(this.ContainingSymbol.IsContainingSymbolOfAllTypeParameters(ImmutableArray<TypeSymbol>.CreateFrom(this.Interfaces)));
-                        Debug.Assert(this.ContainingSymbol.IsContainingSymbolOfAllTypeParameters(this.EffectiveBaseClassNoUseSiteDiagnostics));
-                        Debug.Assert(this.ContainingSymbol.IsContainingSymbolOfAllTypeParameters(this.DeducedBaseTypeNoUseSiteDiagnostics));
+
+                        // PROTOTYPE(static) depending on the resolution for test ExtensionMemberLookup_MatchingExtendedType_GenericMember_TypeOnlyContext_InvalidGenericExtension
```

Comment:
:memo: let's discuss this. The problem is that an invalid extension such as `implicit extension E<T> for object { public static int M; }` can be used with `object.M`. If we use this in a type parameter constraint, then we're holding an instance of `E<T>.M` where `T` is unsubstituted.